### PR TITLE
[Mark All HH] Update household account's indicator on Contact update

### PR DESCRIPTION
**Pull request by npsp-reedestockton**
# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# Features Intended for Future Release

# Features for Elevate Customers

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: force-app/main/domain/LegacyHouseholds.cls [hunk 1]
```diff
@@ -286,9 +286,11 @@ public inherited sharing class LegacyHouseholds extends fflib_SObjects {
         List<Id> householdIdsToUpdate =
                 contactService.moveOppsForContactsSwitchingAccounts(contactsInstance);
         householdService.updateHouseholds(householdIdsToUpdate, dmlWrapper);
+
         Households.updateNameAndMemberCount(
-                contactsInstance.getHouseholdIdsToRename());
-       updateOwners(contactsInstance.getNewOwnerContactById(), dmlWrapper);
+            contactsInstance.getHouseholdIdsForContactsWithDeceasedUpdates(contactsInstance.getHouseholdIdsToRename()));
```
Comment (lparrott) [inline force-app/main/domain/LegacyHouseholds.cls:291]: I see that this is additive, but the way it's written it feels like it's further filtering the Contact list that's being passed (which we wouldn't want to do). Might just be a code organization thing.

Comment (npsp-reedestockton) [reply force-app/main/domain/LegacyHouseholds.cls:291]: Would it help to change the method name to addHouseholdIdsForContactsWithDeceasedUpdates() or do you believe additional code reconfiguration would be a good idea?

Comment (npsp-reedestockton) [reply force-app/main/domain/LegacyHouseholds.cls:291]: Forget that suggestion.  Let's get the Ids and the merge in a separate method.  That should clarify what is happening.

File: force-app/main/domain/LegacyHouseholds.cls [hunk 2]
```diff
@@ -286,9 +286,10 @@ public inherited sharing class LegacyHouseholds extends fflib_SObjects {
         List<Id> householdIdsToUpdate =
                 contactService.moveOppsForContactsSwitchingAccounts(contactsInstance);
         householdService.updateHouseholds(householdIdsToUpdate, dmlWrapper);
-        Households.updateNameAndMemberCount(
-                contactsInstance.getHouseholdIdsToRename());
-       updateOwners(contactsInstance.getNewOwnerContactById(), dmlWrapper);
+
+        Households.updateNameAndMemberCount(contactsInstance.getHouseholdIdsToRenameAndAggregate());
```
[the remark this answers is not in the document]
Comment (lparrott) [reply force-app/main/domain/LegacyHouseholds.cls:290]: I think it works fine, my only other thought is: `getHouseholdIdsToRenameAndCountMembers` ?

[the remark this answers is not in the document]
Comment (npsp-reedestockton) [reply force-app/main/domain/LegacyHouseholds.cls:290]: Yeah, that's a possibility.  My only reservation is that it doesn't reference the major change we've made to how it operates.  If we try to add everything, though, the name would get ridiculous.  getHouseholdIdsToRenameAndCountMembersAndMaybeMarkAllMembersDeceased.  ;-)

**Comment by npsp-reedestockton** [comment]
@lparrott @dfuller114 Ready for review.  I would appreciate your suggestions.  Thanks!

**Comment by npsp-reedestockton** [comment]
Track [W-10342768](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00000ZSKHbYAP/view)

**Review by lparrott — APPROVED** [review]

**Review by dfuller114 — APPROVED** [review]
looks good Reede. great work!

**Pull request merged**